src/WebJobs.Script/Host/ScriptJobHostExtensions.cs

```diff
@@ -19,5 +23,52 @@ public static FunctionDescriptor GetFunctionOrNull(this IScriptJobHost scriptJob
         {
             return scriptJobHost.Functions.FirstOrDefault(p => string.Equals(p.Name, name, StringComparison.OrdinalIgnoreCase));
         }
+
+        /// <summary>
+        /// Lookup a warmup function
+        /// </summary>
+        /// <returns>Warmup function or null if not found</returns>
+        public static FunctionDescriptor GetWarmupFunctionOrNull(this IScriptJobHost scriptJobHost)
+        {
+            var function = scriptJobHost.GetFunctionOrNull(WarmupFunctionName);
+            if (function != null)
+            {
+                bool isWarmupTrigger = function
+                    .Metadata
+                    .InputBindings
+                    .Where(b => b.IsTrigger)
+                    .Any(b => b.Type.Equals(WarmupTriggerName, StringComparison.OrdinalIgnoreCase));
+
+                if (isWarmupTrigger)
+                {
+                    return function;
+                }
+            }
+
```

Comment:
Perhaps go with this approach:

``` csharp
 /// <summary>
        /// Lookup a function by name
        /// </summary>
        /// <param name="name">name of function</param>
        /// <returns>function or null if not found</returns>
        public static FunctionDescriptor GetFunctionOrNull(this IScriptJobHost scriptJobHost, string name)
        {
            return scriptJobHost.Functions.FirstOrDefault(f => IsFunctionNameMatch(f.Name, name));
        }

        public static bool IsFunctionNameMatch(string functionName, string comparison)
        {
            return string.Equals(functionName, comparison, StringComparison.OrdinalIgnoreCase);
        }

        /// <summary>
        /// Lookup a warmup function
        /// </summary>
        /// <returns>Warmup function or null if not found</returns>
        public static FunctionDescriptor GetWarmupFunctionOrNull(this IScriptJobHost scriptJobHost)
        {
            return scriptJobHost.Functions.FirstOrDefault(f =>
            {
                return IsFunctionNameMatch(f.Name, WarmupFunctionName)
                && f.Metadata
                    .InputBindings
                    .Any(b => b.IsTrigger && b.Type.Equals(WarmupTriggerName, StringComparison.OrdinalIgnoreCase));
            });
        }
```

Reply:
Thanks Fabio! That does look a lot cleaner :)